### PR TITLE
Changes to Actions and Middleware

### PR DESCRIPTION
**Pull request by SunlightBro**
added afterReduce() to Actions
Middleware cna no longer make changes to the state
adjusted example and test files to these changes

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -11,4 +11,6 @@ dependencies:
   rxdart: ^0.20.0
 
 dev_dependencies:
+  flutter_test:
```
Comment (jonasbark) [inline pubspec.yaml]: warum das?

**Comment by jonasbark** [comment]
readd tests and adjust them

File: pubspec.yaml (hunk 1 above)
Comment (SunlightBro) [reply pubspec.yaml]: Ohne diese dependency funktionierte flutter test nie.

**Comment by jonasbark** [comment]
Readme(s) have to be adjusted as well?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+# 1.2.0
+* **Breaking** Middlewares does not modifie the state
```
Comment (jonasbark) [inline CHANGELOG.md]: modify

**Pull request merged**